flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -70,6 +73,9 @@ public void write(RowData row) throws IOException {
     switch (row.getRowKind()) {
       case INSERT:
       case UPDATE_AFTER:
+        if (upsert) {
```

Comment:
It seems like this should only happen for the `INSERT` case because `UPDATE_AFTER` implies that there was an `UPDATE_BEFORE` that will perform the delete. This would delete the same row twice in that case, causing more equality deletes to be written for the row.